WordPressKit/ServiceRemoteWordPressComREST.h

```diff
@@ -23,6 +24,12 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, strong, readonly) WordPressComRestApi *wordPressComRestApi;
 
+/**
+ *  @brief      The interface to the WordPress.com API to use for performing REST requests.
+ *              This is meant to gradually replace `wordPressComRestApi`.
+ */
+@property (nonatomic, strong, readonly) id<WordPressComRESTAPIInterfacing> wordPressComRESTAPI;
```

Comment:
The aim of this PR is to get feedback on the direction. As such, I haven't gone through the considerable effort of replacing all usages of the concrete `WordPressComRestApi *wordPressComRestApi` with the abstracted `id<WordPressComRESTAPIInterfacing> wordPressComRESTAPI`.

Assuming the direction I envisioned resonates, that'll be my next step.